src/sonic-build-hooks/scripts/post_run_buildinfo

```diff
@@ -1,16 +1,31 @@
 #!/bin/bash
+set -x
+
+IMAGENAME=$1
 
 . /usr/local/share/buildinfo/scripts/buildinfo_base.sh
 
+
+[ -d $POST_VERSION_PATH ] && rm -rf $POST_VERSION_PATH
+
 # Collect the version files
 collect_version_files $POST_VERSION_PATH
 
-[ -d $BUILD_VERSION_PATH ] && [ ! -z "$(ls -A $BUILD_VERSION_PATH)" ] && cp -rf $BUILD_VERSION_PATH/* $POST_VERSION_PATH
-rm -rf $BUILD_VERSION_PATH/*
-
-# Disable the build hooks
-symlink_build_hooks -d
-set_reproducible_mirrors -d
+if [ ! -z "$(get_version_cache_option)" ]; then
+	# Restore he deletion of cache files
+	cat <<-EOF >/etc/apt/apt.conf.d/docker-clean
+	DPkg::Post-Invoke { "rm -f /var/cache/apt/archives/*.deb /var/cache/apt/archives/partial/*.deb /var/cache/apt/*.bin || true"; };
+	APT::Update::Post-Invoke { "rm -f /var/cache/apt/archives/*.deb /var/cache/apt/archives/partial/*.deb /var/cache/apt/*.bin || true"; };
+	EOF
+	
+fi
+	
+# After the docker build, save the cache contents as tar file and send them to host through log file.
+# Cache contents are encoded as base64 format to avoid the log parsing in the host.
+tar -C ${PKG_CACHE_PATH} --exclude=cache.tgz -zcvf ${PKG_CACHE_FILE_NAME} .
+set +x
+if [[ ! ${IMAGENAME} =~ host-image ]]; then
+	sleep 1;echo  -e "\n_VCSTART_"; (cat ${PKG_CACHE_FILE_NAME} | base64); echo -e "_VCEND_\n";sleep 1
+fi
```

Comment:
It looks like a workaround to transfer the binary files by console log. Are there any other ways to do it? For example, by SSH to the host (or slave container). But it has dependency on SSH, maybe not good enough.

Reply:
I split them into two docker files as you suggested. Could you just take look at the sequence and let me know your thoughts?

# cat Dockerfile
	# Base docker build
	FROM debian:buster
	RUN dd if=/dev/zero of=/root/cache.data bs=100M count=1
	RUN echo $(date)

# cat Dockerfile.cleanup
    # Base docker build
    FROM sonic-image:latest

    # Copy the cache data to host
    From scratch as output
    COPY --from=sonic-image:latest /root/cache.data cache.data

    # Clean up the cache data
    FROM sonic-image:latest as final
    RUN rm /root/cache.data

# cat build.sh
#!/bin/bash
	set -x
	rm -rf out; docker rmi sonic-image-tmp:latest sonic-image:latest

	docker build  --no-cache --tag sonic-image:latest .;
	docker tag sonic-image:latest sonic-image-tmp:latest
	DOCKER_BUILDKIT=1 docker build -f Dockerfile.cleanup  --target output -o out .;
	DOCKER_BUILDKIT=1 docker build -f Dockerfile.cleanup  --no-cache --target final --tag sonic-image:latest .
	docker rmi sonic-image-tmp:latest

	find out; docker images |grep "sonic-image.*latest"

OUTPUT:
====================
+ rm -rf out
+ docker rmi sonic-image-tmp:latest sonic-image:latest
Untagged: sonic-image:latest
Deleted: sha256:7fbc4abec8831240c9d1c574285fdd11b5c47a758d9e957a647ec5271bc750b0
Error: No such image: sonic-image-tmp:latest
+ docker build --no-cache --tag sonic-image:latest .
Sending build context to Docker daemon  8.192kB
Step 1/3 : FROM debian:buster
 ---> 4a7a1f401734
Step 2/3 : RUN dd if=/dev/zero of=/root/cache.data bs=100M count=1
 ---> Running in 0fe5524ba638
1+0 records in
1+0 records out
104857600 bytes (105 MB, 100 MiB) copied, 0.129012 s, 813 MB/s
Removing intermediate container 0fe5524ba638
 ---> 76fea604fe86
Step 3/3 : RUN echo $(date)
 ---> Running in 4aa14234d98e
Wed Nov 23 13:46:58 UTC 2022
Removing intermediate container 4aa14234d98e
 ---> 07028c1daa48
Successfully built 07028c1daa48
Successfully tagged sonic-image:latest
+ docker tag sonic-image:latest sonic-image-tmp:latest
+ DOCKER_BUILDKIT=1
+ docker build -f Dockerfile.cleanup --target output -o out .
[+] Building 0.6s (5/5) FINISHED
 => [internal] load build definition from Dockerfile.cleanup                                                                                                                    0.0s
 => => transferring dockerfile: 377B                                                                                                                                            0.0s
 => [internal] load .dockerignore                                                                                                                                               0.0s
 => => transferring context: 2B                                                                                                                                                 0.0s
 => FROM docker.io/library/sonic-image:latest                                                                                                                                   0.0s
 => => resolve docker.io/library/sonic-image:latest                                                                                                                             0.0s
 => CACHED [output 1/1] COPY --from=sonic-image:latest /root/cache.data cache.data                                                                                              0.0s
 => exporting to client                                                                                                                                                         0.3s
 => => copying files 104.88MB                                                                                                                                                   0.3s
+ DOCKER_BUILDKIT=1
+ docker build -f Dockerfile.cleanup --no-cache --target final --tag sonic-image:latest .
[+] Building 0.6s (6/6) FINISHED
 => [internal] load build definition from Dockerfile.cleanup                                                                                                                    0.0s
 => => transferring dockerfile: 46B                                                                                                                                             0.0s
 => [internal] load .dockerignore                                                                                                                                               0.0s
 => => transferring context: 2B                                                                                                                                                 0.0s
 => [internal] load metadata for docker.io/library/sonic-image:latest                                                                                                           0.0s
 => CACHED [final 1/2] FROM docker.io/library/sonic-image:latest                                                                                                                0.0s
 => [final 2/2] RUN rm /root/cache.data                                                                                                                                         0.6s
 => exporting to image                                                                                                                                                          0.0s
 => => exporting layers                                                                                                                                                         0.0s
 => => writing image sha256:b7337d033d1088956a563ea2b111f467ba015566b3afb9b07b0b5964113699d5                                                                                    0.0s
 => => naming to docker.io/library/sonic-image:latest                                                                                                                           0.0s
+ docker rmi sonic-image-tmp:latest
Untagged: sonic-image-tmp:latest
Deleted: sha256:07028c1daa486d4adbcd1fb0ae4ebf634f80f0fba1c204fd766d918e9ccc8a52
Deleted: sha256:76fea604fe863f93a94fac7b608ac794598769369601995bac5d0d777d6c9e5e
+ find out
out
out/cache.data
+ docker images
+ grep 'sonic-image.*latest'
sonic-image                               latest              b7337d033d10        1 second ago        219MB

Reply:
The solution looks good for me, thanks @Kalimuthu-Velappan